shared/src/utils/validators/is-uuid-set.ts

```diff
@@ -0,0 +1,20 @@
+import type { ValidationOptions, ValidationArguments } from 'class-validator';
+import { isUUID } from 'class-validator';
+import { getMapValidator } from './get-map-validator';
+import { makeValidator } from './make-validator';
+
+export function isUUIDSet(valueToBeValidated: unknown): boolean {
+    return getMapValidator<true>({
+        keyValidator: (key) => isUUID(key, 4),
+        valueValidator: (value) => value === true,
+    })(valueToBeValidated);
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export function IsUUIDSet(validationOptions?: ValidationOptions) {
+    return makeValidator(
+        'isUUIDSet',
+        (value: unknown, args?: ValidationArguments) => isUUIDSet(value),
+        validationOptions
+    );
+}
```

Comment:
Wouldn't it be more performant to do it like this:

```suggestion
const isUUIDSet = getMapValidator<true>({
    keyValidator: (key) => isUUID(key, 4),
    valueValidator: (value) => value === true,
});

// eslint-disable-next-line @typescript-eslint/naming-convention
export function IsUUIDSet(validationOptions?: ValidationOptions) {
    return makeValidator(
        'isUUIDSet',
        (value: unknown, args?: ValidationArguments) => isUUIDSet(value),
        validationOptions
    );
}
```

Reply:
the same for the other validators.